### PR TITLE
Add advanced resolver modes (passthrough, auto_convert, prefix, aliases)

### LINKED ISSUE #3: Add advanced resolver modes (passthrough, auto_convert, prefix, aliases)
## Description
Enhance the value resolver to support multiple resolution modes like hashids-bundle.

## Tasks
- [ ] Add `passthrough` configuration option - when enabled, sets the decoded value in request attributes for the next resolver (useful for Doctrine entity conversion)
- [ ] Add `auto_convert` configuration option - when enabled, automatically attempts to decode all string parameters
- [ ] Add prefix support (`_sqid_`) - parameters prefixed with `_sqid_` are explicitly decoded
- [ ] Add alias support for `sqid` and `id` parameter names
- [ ] Update `Configuration.php` with new options
- [ ] Update `SqidsValueResolver` with the new logic
- [ ] Add comprehensive specs for all modes

## Configuration Example
```yaml
sqids:
    passthrough: false
    auto_convert: false
```

## Reference
See hashids-bundle implementation: `src/ValueResolver/HashidsValueResolver.php`

### PR DESCRIPTION
**Pull request by roukmoute**
## Summary
- Add `passthrough` config: sets decoded value in request attributes for next resolver (useful for Doctrine entity conversion)
- Add `auto_convert` config: automatically decode all string parameters
- Add prefix support: `_sqid_` prefix for explicit conversion
- Add alias support: `sqid` and `id` parameter names are automatically resolved
- Update Configuration.php with new options
- Update SqidsValueResolver with new logic
- Add comprehensive specs for all modes

## Configuration
```yaml
roukmoute_sqids:
    alphabet: 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789'
    min_length: 0
    blocklist: null
    passthrough: false  # NEW
    auto_convert: false # NEW
```

### Prefix support
```php
// Route: /users/{_sqid_id}
public function show(int $id): Response
{
    // $id is decoded from the _sqid_id route parameter
}
```

### Alias support
```php
// Route: /users/{id} or /users/{sqid}
public function show(int $userId): Response
{
    // $userId is decoded from 'id' or 'sqid' route parameter
}
```

### Auto convert mode
```yaml
roukmoute_sqids:
    auto_convert: true
```
```php
// Route: /users/{id}
public function show(int $id): Response
{
    // $id is automatically decoded if it looks like a sqid
}
```

### Passthrough mode (for Doctrine)
```yaml
roukmoute_sqids:
    passthrough: true
```
```php
// Route: /users/{id}
public function show(User $user): Response
{
    // The sqid is decoded and set in request attributes,
    // then Doctrine's ParamConverter resolves the User entity
}
```

Closes #3

### CODE REVIEW

File: src/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -38,6 +38,14 @@ public function getConfigTreeBuilder(): TreeBuilder
                     ->info('A list of sqids to block')
                     ->defaultValue(null)
                 ->end()
+                ->booleanNode('passthrough')
+                    ->info('If true, sets decoded value in request attributes for the next resolver (useful for Doctrine entity conversion)')
+                    ->defaultFalse()
+                ->end()
+                ->booleanNode('auto_convert')
+                    ->info('If true, automatically attempts to decode all string parameters. Use "_sqid_" prefix in routing for explicit conversion.')
```
[the remark this answers is not in the document]
Comment (roukmoute) [reply src/DependencyInjection/Configuration.php]: Fixed: Updated the info text to clarify that auto_convert decodes route parameters matching int argument names, and that _sqid_ prefix and #[Sqid] attribute work regardless of this setting.

File: src/ValueResolver/SqidsValueResolver.php [hunk 2]
```diff
@@ -5,76 +5,79 @@
...
 class SqidsValueResolver implements ValueResolverInterface
 {
-    public function __construct(private Sqids $sqids)
-    {
+    public function __construct(
+        private readonly SqidsInterface $sqids,
+        private readonly bool $passthrough,
+        private readonly bool $autoConvert,
+        private readonly string $alphabet,
+    ) {
```
[the remark this answers is not in the document]
Comment (roukmoute) [reply src/ValueResolver/SqidsValueResolver.php:20]: Fixed: Added test `it_resolves_with_passthrough_and_auto_convert_enabled` to cover this scenario.

File: src/ValueResolver/SqidsValueResolver.php [hunk 3]
```diff
@@ -86,22 +89,62 @@ private function getSqidAttribute(ArgumentMetadata $argument): ?Sqid
...
+        if (!$request->attributes->has('sqids_prevent_alias')) {
+            foreach (['sqid', 'id'] as $alias) {
+                if ($request->attributes->has($alias)) {
+                    $aliasAttribute = $request->attributes->get($alias);
+                    if (!\is_string($aliasAttribute)) {
+                        continue;
+                    }
+                    $sqid = $aliasAttribute;
+                    $request->attributes->set('sqids_prevent_alias', true);
+                    break;
```
[the remark this answers is not in the document]
Comment (roukmoute) [reply src/ValueResolver/SqidsValueResolver.php:133]: Fixed: Added tests `it_skips_when_decode_returns_empty_array` and `it_throws_when_decode_returns_empty_array_with_attribute` to cover this edge case.

[the remark this answers is not in the document]
Comment (roukmoute) [reply src/ValueResolver/SqidsValueResolver.php:133]: Already handled: The `isValidDecode` method checks `count($decoded) > 1` and throws LogicException when explicit, or returns false to skip silently otherwise.

File: src/ValueResolver/SqidsValueResolver.php [hunk 4]
```diff
@@ -5,76 +5,79 @@
...
-        $value = $request->attributes->get($routeParameter);
+        [$sqid, $isExplicit] = $this->getSqid($request, $routeParameter, $sqidAttribute !== null);
 
-        if ($argument->isVariadic() || !\is_string($value)) {
+        if ($this->isSkippable($sqid)) {
             return [];
         }
 
-        $hasAttribute = $sqidAttribute !== null;
+        $decoded = $this->sqids->decode($sqid);
```
[the remark this answers is not in the document]
Comment (roukmoute) [reply src/ValueResolver/SqidsValueResolver.php:41]: Fixed: Changed to use `$decoded[0]` with explicit empty array check `$decoded === []` in `isValidDecode`.

File: src/ValueResolver/SqidsValueResolver.php [hunk 5]
```diff
@@ -5,76 +5,79 @@
...
-        $sqidAttribute = $this->getSqidAttribute($argument);
+        if ($argument->isVariadic()) {
+            return [];
+        }
+
         $name = $argument->getName();
+        $sqidAttribute = $this->getSqidAttribute($argument);
         $routeParameter = $sqidAttribute?->parameter ?? $name;
-        $value = $request->attributes->get($routeParameter);
+        [$sqid, $isExplicit] = $this->getSqid($request, $routeParameter, $sqidAttribute !== null);
```
[the remark this answers is not in the document]
Comment (roukmoute) [reply src/ValueResolver/SqidsValueResolver.php:35]: This comment is outdated: The variadic check IS present at lines 28-30. The test `it_fails_to_resolve_variadic_argument` passes.

File: src/ValueResolver/SqidsValueResolver.php [hunk 6]
```diff
@@ -5,76 +5,79 @@
...
+        if ($this->isSkippable($sqid)) {
             return [];
         }
 
-        $hasAttribute = $sqidAttribute !== null;
+        $decoded = $this->sqids->decode($sqid);
 
-        if (!$hasAttribute && !$this->isValidType($argument->getType())) {
+        if (!$this->isValidDecode($decoded, $isExplicit, $name)) {
             return [];
```
[the remark this answers is not in the document]
Comment (roukmoute) [reply src/ValueResolver/SqidsValueResolver.php:44]: By design: In passthrough mode, the decoded value is set using the argument name ($name) because the next resolver (typically Doctrine's EntityValueResolver) expects the attribute to match the controller argument name. The route parameter name and argument name are typically the same (e.g., `/users/{id}` with `int $id`), or when different, the #[Sqid(parameter: 'routeParam')] attribute can be used to specify the source.

File: src/ValueResolver/SqidsValueResolver.php [hunk 7]
```diff
@@ -5,76 +5,79 @@
...
-            if ($class) {
-                $request->attributes->set($name, $decode[0]);
+            return false;
+        }
 
-                return [];
+        if (\count($decoded) > 1) {
+            if ($isExplicit) {
+                throw new \LogicException(sprintf('Unable to decode parameter "%s".', $name));
             }
```
[the remark this answers is not in the document]
Comment (roukmoute) [reply src/ValueResolver/SqidsValueResolver.php:76]: Edge case acknowledged but unlikely in practice: Naming a route parameter with `_sqid_` prefix (e.g., `{_sqid_id}`) is not a documented or expected usage pattern. The prefix is meant to be used in route definitions to signal explicit sqid conversion, not as part of the actual parameter name. This would be a misuse of the convention.

File: src/ValueResolver/SqidsValueResolver.php [hunk 8]
```diff
@@ -86,22 +89,62 @@ private function getSqidAttribute(ArgumentMetadata $argument): ?Sqid
...
 
-        return [$decodedValue];
+        return ['', false];
+    }
+
+    private function getSqidFromAliases(Request $request): string
+    {
+        $sqid = '';
+
+        if (!$request->attributes->has('sqids_prevent_alias')) {
```
[the remark this answers is not in the document]
Comment (roukmoute) [reply src/ValueResolver/SqidsValueResolver.php:124]: Not an issue: `preg_quote()` with the delimiter parameter correctly escapes any characters that would conflict with the regex syntax, including curly braces. The delimiter `{` is explicitly passed to `preg_quote()` so it knows to escape it if present in the alphabet. This is the standard PHP approach for building dynamic regex patterns.

File: src/ValueResolver/SqidsValueResolver.php [hunk 9]
```diff
@@ -5,76 +5,79 @@
...
+        /** @var int $decodedValue */
+        $decodedValue = reset($decoded);
 
-    private function isValidType(?string $type): bool
-    {
-        return $type === 'int' || ($type !== null && class_exists($type));
+        if ($this->passthrough) {
+            $request->attributes->set($name, $decodedValue);
+
+            return [];
```
[the remark this answers is not in the document]
Comment (roukmoute) [reply src/ValueResolver/SqidsValueResolver.php:53]: Won't fix for now: The error message is intentionally concise. Including the actual sqid value could expose sensitive data in logs. The parameter name is sufficient for debugging. If more context is needed in the future, this can be revisited, but for a 1.0 release this is acceptable.

File: src/ValueResolver/SqidsValueResolver.php (hunk 8 above)
[the remark this answers is not in the document]
Comment (roukmoute) [reply src/ValueResolver/SqidsValueResolver.php:124]: Not an issue: `preg_quote()` handles all special regex characters including those with special meaning in character classes (`]`, `^`, `-`, `\`). The function escapes them with backslashes, making them literal characters. This is well-tested PHP behavior. Additionally, Sqids alphabets are validated by the Sqids library itself and typically contain only alphanumeric characters.

File: src/ValueResolver/SqidsValueResolver.php [hunk 10]
```diff
@@ -86,22 +89,62 @@ private function getSqidAttribute(ArgumentMetadata $argument): ?Sqid
...
+        $sqid = $this->getSqidFromAliases($request);
+        if ($sqid !== '') {
+            return [$sqid, true];
         }
 
-        return [$decodedValue];
+        return ['', false];
+    }
+
+    private function getSqidFromAliases(Request $request): string
```
[the remark this answers is not in the document]
Comment (roukmoute) [reply src/ValueResolver/SqidsValueResolver.php:120]: By design: The `isSkippable()` method intentionally performs a lightweight pre-validation using alphabet character matching before calling the actual decode. This prevents unnecessary decode attempts on values that are clearly not sqids (e.g., UUIDs, slugs, numeric strings). The Sqids library handles invalid sqids gracefully by returning an empty array, which is then caught by `isValidDecode()`.

File: src/ValueResolver/SqidsValueResolver.php [hunk 11]
```diff
@@ -5,76 +5,79 @@
...
 
-            if ($class) {
-                $request->attributes->set($name, $decode[0]);
+            return false;
+        }
 
-                return [];
+        if (\count($decoded) > 1) {
+            if ($isExplicit) {
+                throw new \LogicException(sprintf('Unable to decode parameter "%s".', $name));
```
[the remark this answers is not in the document]
Comment (roukmoute) [reply src/ValueResolver/SqidsValueResolver.php:74]: Not an issue: `ArgumentMetadata::getName()` returns `string` according to Symfony's type declarations, not `?string`. The empty string check handles the edge case where the argument name might be empty (though also unlikely). This is defensive coding without over-engineering.

File: src/ValueResolver/SqidsValueResolver.php [hunk 12]
```diff
@@ -86,22 +89,62 @@ private function getSqidAttribute(ArgumentMetadata $argument): ?Sqid
...
+        $sqid = $request->attributes->get('_sqid_' . $name);
+        if (isset($sqid) && \is_string($sqid)) {
+            return [$sqid, true];
         }
 
-        $decodedValue = $decodedValues[0];
+        if ($this->autoConvert || $hasSqidAttribute) {
+            $sqid = $request->attributes->get($name);
+            if (\is_string($sqid)) {
+                return [$sqid, $hasSqidAttribute];
```
[the remark this answers is not in the document]
Comment (roukmoute) [reply src/ValueResolver/SqidsValueResolver.php:108]: By design: The `sqids_prevent_alias` flag is intentionally request-scoped. Its purpose is to prevent the same alias (`sqid` or `id`) from being used to resolve multiple controller arguments within the same request. Once an alias is consumed by one argument, other arguments should not implicitly reuse it. This matches the behavior of hashids-bundle and prevents ambiguous resolution scenarios.

File: src/ValueResolver/SqidsValueResolver.php [hunk 13]
```diff
@@ -5,76 +5,79 @@
...
     /**
      * @return iterable<int>
      */
     public function resolve(Request $request, ArgumentMetadata $argument): iterable
     {
-        $sqidAttribute = $this->getSqidAttribute($argument);
+        if ($argument->isVariadic()) {
+            return [];
+        }
+
```
[the remark this answers is not in the document]
Comment (roukmoute) [reply src/ValueResolver/SqidsValueResolver.php:31]: By design: When using `#[Sqid(parameter: 'custom')]`, the resolver checks for `_sqid_custom` first, then falls back to `custom`. This is intentional - it allows explicit prefix usage even with custom parameter names. The priority order is: 1) `_sqid_` prefixed attribute, 2) direct parameter match, 3) aliases. This provides maximum flexibility while maintaining backward compatibility.
